scripts/ci/changes.js

```diff
@@ -0,0 +1,14 @@
+import { GitUtils } from "@backstage/cli-node";
+
+const changedFiles = await GitUtils.listChangedFiles("origin/main");
+const workspaces = new Set();
+
+changedFiles.forEach((name) => {
+  // First match with prefix = "workspaces/" & suffix = "/"
+  const match = name.match("(?<=workspaces/)([^/]+)(?=/)");
```

Comment:
```suggestion
  const match = name.match("workspaces/([^/]+)/");
```

No need for the fancy post-and prefix matching right? :)

Also, i don't know if these are relative or absolute, but it would feel better if we could anchor it with `^` at the start so it doesn't spuriously match on these words later in the path somewhere. Or `string.split` or whatnot

And speaking of that ... does this work on windows too? Will we have windows based tests later? Does the GitUtils always emit forward slashes or not

Reply:
it shouldn't matter so much as this script is only meant to be run by the CI on an ubuntu runner